### DISCUSSION TITLE
Pressure regulator for pipe flow calculation

### DISCUSSION THREAD

**Discussion by FrancescaSoldan-RSE** [e2nIEE/pandapipes#600] [Q&A]
Hi, I am trying to model a pressure regulator in a gas network. 
I have read on this paper (https://publica-rest.fraunhofer.de/server/api/core/bitstreams/72a583a8-7204-4f87-bc9d-10760e07701a/content) that regulators have been added to the pandapipes library. However, I can not find a way to correctly model them in a gas distribution network, in order to find the correct results of gas flow calculations (I tried with compressors or valves between junctions).
Do you have any ideas of how to model this component?
Thank you in advance

**Comment by dlohmeier** [comment]
Hi @FrancescaSoldan-RSE ,
there is a pressure control component implemented in pandapipes ready to use (see [here](https://github.com/e2nIEE/pandapipes/blob/2883f24269df7330a8dc03be9c28af82a993c364/src/pandapipes/create.py#L885)). You can just use the function `pandapipes.create_pressure_control(net, ...)` . If you still experience any problems, just let us know.
Best regards and good luck.

**Reply by FrancescaSoldan-RSE** [reply]
Thank you @dlohmeier for the suggestion!
Regards.

**Reply by FrancescaSoldan-RSE** [reply]
Hi @uvicorn666, as suggested by @dlohmeier, the pressure control component appears to be functioning correctly. It takes nominal pressures from the start and end junctions, while the fixed regulated pressure is specified through the controlled_junction and controlled_p_bar parameters. In this way, I am effectively obtaining the right results for gas flow calculations. 
Regards, Francesca.
